### PR TITLE
Tidy nested namespaces

### PR DESCRIPTION
**Pull request by mjcarroll**
Part of https://github.com/gazebosim/gz-sim/issues/2276

* Condense nested namespaces and remove "using namespace" where possible: [177c775](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/177c7758a35e69473591c5cd87fae24991212c4a)
* Split some larger headers into corresponding pieces: [a02c156](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/a02c156b639758f59da02fa15f98701b63fdc0f7)
* Split out some of the types that were defined in larger headers: [097ce9e](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/097ce9ea4b585c5fdc3cff8651f9902e12f116d2)

@j-rivero I removed some `TODOs` about renaming the `ResultType::DELETE` and `HttpMethod::DELETE` entries.  I think that keeping the `DELETE` name is the most appropriate with the `undefs`.  Since we are now using it purely in `enum class` and the other use is part of the [standard access rights](https://learn.microsoft.com/en-us/windows/win32/secauthz/standard-access-rights?redirectedfrom=MSDN) I think that the chance of collision is low enough to not worry about.

### CODE REVIEW

File: include/gz/fuel_tools/HttpMethod.hh [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2017 Open Source Robotics Foundation
```
Comment (ahcorde) [inline include/gz/fuel_tools/HttpMethod.hh:2]: ?
```suggestion
 * Copyright (C) 2023 Open Source Robotics Foundation
```

File: include/gz/fuel_tools/ResultType.hh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef GZ_FUEL_TOOLS_RESULTTYPE_HH_
+#define GZ_FUEL_TOOLS_RESULTTYPE_HH_
+
+#include <memory>
+#include <string>
```
Comment (ahcorde) [inline include/gz/fuel_tools/ResultType.hh:22]: remove

File: include/gz/fuel_tools/ResultType.hh [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2017 Open Source Robotics Foundation
```
Comment (ahcorde) [inline include/gz/fuel_tools/ResultType.hh:2]: ?
```suggestion
 * Copyright (C) 2024 Open Source Robotics Foundation
```

File: include/gz/fuel_tools/ServerConfig.hh [hunk 4]
```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2017 Open Source Robotics Foundation
```
Comment (ahcorde) [inline include/gz/fuel_tools/ServerConfig.hh:2]: ?
```suggestion
 * Copyright (C) 2024 Open Source Robotics Foundation
```

File: include/gz/fuel_tools/ServerConfig.hh [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+ * limitations under the License.
+ *
+*/
+
+#ifndef GZ_FUEL_TOOLS_SERVERCONFIG_HH_
+#define GZ_FUEL_TOOLS_SERVERCONFIG_HH_
+
+#include <memory>
+#include <string>
+#include <vector>
```
Comment (ahcorde) [inline include/gz/fuel_tools/ServerConfig.hh:23]: remove

File: src/ModelIterPrivate.hh [hunk 6]
```diff
@@ -33,155 +33,152 @@
...
+
+    /// \brief Model identifiers in the current page
+    protected: std::vector<ModelIdentifier> ids;
+
+    /// \brief Where the current iterator is in the list of ids
+    protected: std::vector<ModelIdentifier>::iterator idIter;
+
+    /// \brief Keep track of page number for pagination of response data from
+    /// server.
+    protected: std::size_t currentPage{0};
```
Comment (ahcorde) [inline src/ModelIterPrivate.hh:176]: include `<cstddef>`

File: src/ServerConfig.cc [hunk 7]
```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2017 Open Source Robotics Foundation
```
Comment (ahcorde) [inline src/ServerConfig.cc:2]: ?
```suggestion
 * Copyright (C) 2024 Open Source Robotics Foundation
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
Some minor comments

File: src/ServerConfig_TEST.cc [hunk 8]
```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2017 Open Source Robotics Foundation
```
[the remark this answers is not in the document]
Comment (mjcarroll) [reply src/ServerConfig_TEST.cc]: [8a56903](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/8a56903ea1ad1bf4c12e88c0cd78acbd6a0ee673)

File: src/ServerConfig.cc (hunk 7 above)
Comment (mjcarroll) [reply src/ServerConfig.cc]: [8a56903](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/8a56903ea1ad1bf4c12e88c0cd78acbd6a0ee673)

File: src/ModelIterPrivate.hh (hunk 6 above)
Comment (mjcarroll) [reply src/ModelIterPrivate.hh:176]: [8a56903](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/8a56903ea1ad1bf4c12e88c0cd78acbd6a0ee673)

File: include/gz/fuel_tools/ServerConfig.hh (hunk 5 above)
Comment (mjcarroll) [reply include/gz/fuel_tools/ServerConfig.hh]: [8a56903](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/8a56903ea1ad1bf4c12e88c0cd78acbd6a0ee673)

File: include/gz/fuel_tools/ServerConfig.hh (hunk 4 above)
Comment (mjcarroll) [reply include/gz/fuel_tools/ServerConfig.hh]: [8a56903](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/8a56903ea1ad1bf4c12e88c0cd78acbd6a0ee673)

File: include/gz/fuel_tools/ResultType.hh (hunk 3 above)
Comment (mjcarroll) [reply include/gz/fuel_tools/ResultType.hh]: [8a56903](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/8a56903ea1ad1bf4c12e88c0cd78acbd6a0ee673)

File: include/gz/fuel_tools/ResultType.hh (hunk 2 above)
Comment (mjcarroll) [reply include/gz/fuel_tools/ResultType.hh]: [8a56903](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/8a56903ea1ad1bf4c12e88c0cd78acbd6a0ee673)

File: include/gz/fuel_tools/HttpMethod.hh (hunk 1 above)
Comment (mjcarroll) [reply include/gz/fuel_tools/HttpMethod.hh]: [8a56903](https://github.com/gazebosim/gz-fuel-tools/pull/396/commits/8a56903ea1ad1bf4c12e88c0cd78acbd6a0ee673)

**Review by ahcorde — APPROVED** [review]

**Pull request merged**